app/src/main/kotlin/io/imhungry/common/ui/BaseActivity.kt

```diff
@@ -5,13 +5,15 @@ import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
 import com.google.firebase.auth.FirebaseAuth
 import dagger.android.AndroidInjection
+import io.imhungry.login.AuthFailureCallback
 import io.imhungry.login.handleLoginActivityResult
+import io.imhungry.login.launchLoginActivity
 
 abstract class BaseActivity : AppCompatActivity(), FirebaseAuth.AuthStateListener {
 
-    protected var userUnauthenticatedHandler: (() -> Unit)? = null
```

Comment:
Why did you remove this? It should also be of type `AuthFailureCallback?`, with a default of null.